deeptime/src/include/deeptime/markov/msm/tram/tram_input.h

```diff
@@ -20,12 +20,17 @@ constexpr void throwIfInvalid(bool isValid, const std::string &message) {
 template<typename dtype>
 class TRAMInput {
 public:
-    TRAMInput(CountsMatrix &&stateCounts, CountsMatrix &&transitionCounts,
-              DTraj dtraj, BiasMatrix<dtype> biasMatrix)
+    using size_type = typename BiasMatrices<dtype>::size_type;
+
+    TRAMInput(CountsMatrix &&stateCounts, CountsMatrix &&transitionCounts, BiasMatrices<dtype> biasMatrices)
             : stateCounts_(std::move(stateCounts)),
               transitionCounts_(std::move(transitionCounts)),
-              dtraj_(std::move(dtraj)),
-              biasMatrix_(std::move(biasMatrix)) {
+              biasMatrices_(std::move(biasMatrices)),
+              cumNSamples_() {
+        cumNSamples_.resize(nMarkovStates());
+        for(std::size_t i = 1; i < cumNSamples_.size(); ++i) {
+            cumNSamples_[i] += cumNSamples_[i-1] + nSamples(i-1);
+        }
```

Comment:
[std::partial_sum](https://en.cppreference.com/w/cpp/algorithm/partial_sum) could maybe be used here and perhaps also in the `Index` class